### PR TITLE
Laboratory work #3, Alyona Gosteva - 21FPL2

### PR DESCRIPTION
**Pull request by Gostaa**

### CODE REVIEW

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -101,7 +89,9 @@ def preprocess_text(self, text: str) -> tuple[str, ...]:
             tuple[str, ...]
                 clean lowercase tokens with no stop-words
         """
-        return self._remove_stop_words(self._clean_and_tokenize(text))
+        tokens = self._clean_and_tokenize(text)
+        filtered_tokens = self._remove_stop_words(tokens)
+        return filtered_tokens
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:94]: you can return the result of `_remove_stop_words` without saving to a variable

File: lab_3_keywords_textrank/main.py [hunk 2]
```diff
@@ -141,11 +131,11 @@ def _learn_indices(self, tokens: tuple[str, ...]) -> None:
             tokens : tuple[str, ...]
                 sequence of string tokens
         """
-        for token, idx in zip(tokens, range(1001 + len(tokens), 1001, -1)):
-            self._word2id[token] = idx
-
-        for token, idx in self._word2id.items():
-            self._id2word[idx] = token
+        code = 1000
+        for word in set(tokens):
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:135]: try to use `enumerate `

File: lab_3_keywords_textrank/main.py [hunk 3]
```diff
@@ -623,10 +629,9 @@ def update_vertex_score(self, vertex: int, incidental_vertices: list[int], score
             scores: dict[int, float]
                 scores of all vertices in the graph
         """
-        summa = sum((1 / self._graph.calculate_inout_score(inc_vertex)) * scores[inc_vertex]
-                    for inc_vertex in incidental_vertices)
-        self._scores[vertex] = summa * self._damping_factor + (1 - self._damping_factor)
-        pass
+        self._scores[vertex] = (1 - self._damping_factor) + self._damping_factor * sum(
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:632]: too long for one line

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (Gostaa) [reply lab_3_keywords_textrank/main.py:94]: done

File: lab_3_keywords_textrank/main.py (hunk 3 above)
Comment (Gostaa) [reply lab_3_keywords_textrank/main.py:632]: done

File: lab_3_keywords_textrank/main.py (hunk 2 above)
Comment (Gostaa) [reply lab_3_keywords_textrank/main.py:135]: done

**Comment by Gostaa** [comment]
@artyomtugaryov , I've finished lab for mark 8 on time, but I've decided to continue working for mark 10

File: lab_3_keywords_textrank/main.py [hunk 4]
```diff
@@ -971,45 +981,47 @@ def run(self) -> Optional[dict[str, dict[str, float]]]:
...
+                top_keywords = algorithm.get_top_keywords(50)
+                if name in ('VanillaTextRank', 'PositionBiasedTextRank'):
+                    top_keywords = encoder.decode(top_keywords)
+                recall = calculate_recall(top_keywords, keywords)
+
+                if name not in report:
+                    report[name] = {}
+                report[name][theme] = recall
+        self.report = report
+        return report
```
Comment (demid5111) [inline lab_3_keywords_textrank/main.py:1024]: avg recall print

**Pull request closed without merging**